lib/masamune/abstract_syntax_tree.rb

```diff
@@ -164,18 +107,12 @@ def replace(type:, old_token:, new_token:)
 
     private
 
-    def get_node_class(type)
-      "#{self.class}::#{type.to_s.camelize}".safe_constantize || Node # Return base Node class for any not-yet-covered nodes.
-    end
-
-    # We only order results when they are a Hash.
-    # The contents from the Hash are pulled from data nodes.
-    # i.e. - {position: [4, 7], token: "project"}
-    def order_results(results)
-      if results.first.is_a?(Hash)
-        DataNode.order_results_by_position(results)
-      else
-        results
+    # Order results according to their token location.
+    def order_nodes(nodes)
+      nodes.sort do |a, b|
+        by_line = a.token_location.start_line <=> b.token_location.start_line
+        # If we're on the same line, refer to the inner index for order.
+        by_line.zero? ? a.token_location.start_column <=> b.token_location.start_column : by_line
```

Comment:
I thought Prism nodes were already sorted like this? Since they're just added by how they're structured in the source code.

Reply:
Yes, this mainly has to do with the LALR parsing algorithm in that it takes the `CallNodes` from the right side and moves to the left. This was the same thing I was running into with Ripper originally, and I wanted the nodes to be ordered according to their location, as opposed to the order in which they're called (mainly since we do a lot of line-number-based editing in the [RoutesFileManipulator](https://github.com/bullet-train-co/bullet_train-core/blob/main/bullet_train-super_scaffolding/lib/scaffolding/routes_file_manipulator.rb), etc.).

So, say we have the following code:
```ruby
code = <<~CODE
  [1, 2].sum.odd?
CODE

parsed_code = Prism.parse(code)
p parsed_code.value
```

This will give us the following Prism output:
```
=> @ ProgramNode (location: (1,0)-(1,15))
├── locals: []
└── statements:
    @ StatementsNode (location: (1,0)-(1,15))
    └── body: (length: 1)
        └── @ CallNode (location: (1,0)-(1,15))
            ├── receiver:
            │   @ CallNode (location: (1,0)-(1,10))
            │   ├── receiver:
            │   │   @ ArrayNode (location: (1,0)-(1,6))
            │   │   ├── elements: (length: 2)
            │   │   │   ├── @ IntegerNode (location: (1,1)-(1,2))
            │   │   │   │   └── flags: decimal
            │   │   │   └── @ IntegerNode (location: (1,4)-(1,5))
            │   │   │       └── flags: decimal
            │   │   ├── opening_loc: (1,0)-(1,1) = "["
            │   │   └── closing_loc: (1,5)-(1,6) = "]"
            │   ├── call_operator_loc: (1,6)-(1,7) = "."
            │   ├── message_loc: (1,7)-(1,10) = "sum"
            │   ├── opening_loc: ∅
            │   ├── arguments: ∅
            │   ├── closing_loc: ∅
            │   ├── block: ∅
            │   ├── flags: ∅
            │   └── name: "sum"
            ├── call_operator_loc: (1,10)-(1,11) = "."
            ├── message_loc: (1,11)-(1,15) = "odd?"
            ├── opening_loc: ∅
            ├── arguments: ∅
            ├── closing_loc: ∅
            ├── block: ∅
            ├── flags: ∅
            └── name: "odd?"
```

You can see that the first `CallNode`'s name is `odd?` (all the way at the very bottom), and then the nested `CallNode` is `sum`.

Again, I did this mainly to order by line number/column, but maybe I'm going about it incorrectly? I'm open to ideas.